linetools/isgm/abssightline.py

```diff
@@ -0,0 +1,242 @@
+
+        slf = cls.from_components(comps)
+
+        # Return
+        return slf
+
+    @classmethod
+    def from_components(cls, components, **kwargs):
+        """ Instantiate from a list of AbsComponent objects
+
+        Uses RA/DEC, Zion, Ej, A, z, vlim
+
+        Parameters
+        ----------
+        components : list
+           list of AbsComponent objects
+
+        Returns
+        -------
+        """
+        # Check
+        if not isinstance(components[0], AbsComponent):
+            raise IOError('Need an AbsComponent object')
+
+        # Instantiate with the first component
+        slf = cls( components[0].coord)
+        for comp in components:
+            slf.add_component(comp, **kwargs)
+        # Return
+        return slf
+
+    def __init__(self, radec, sl_type=None, em_type=None, comment=None, name=None):
+        """  Initiator
+
+        Parameters
+        ----------
+        radec : tuple or SkyCoord
+            (RA,DEC) in deg or astropy.coordinate
+        sl_type : str, optional
+          Sightline type, e.g. IGM
+        emtype : str, optional
+          Type of emission source for absorption sightline
+        comment : str, optional
+          A comment, default is ``
+        """
+        # Required
+        self.coord = ltu.radec_to_coord(radec)
+
+        # Lists
+        self._components = []
+
+        # Name
+        if name is None:
+            self.name = 'J{:s}{:s}'.format(
+                    self.coord.ra.to_string(unit=u.hour,sep='',pad=True),
+                    self.coord.dec.to_string(sep='',pad=True,alwayssign=True))
+        else:
+            self.name=name
+
+        # Others
+        self.em_type = em_type
+        self.sl_type = sl_type
+
+    def add_component(self, abscomp, tol=0.2*u.arcsec,
+                      chk_sep=True, debug=False, **kwargs):
+        """Add an AbsLine object to the component if it satisfies
+        all of the rules.
+
+        For velocities, we demand that the new line has a velocity
+        range that is fully encompassed by the component.
+
+        Parameters
+        ----------
+        abscomp : AbsComp
+        tol : Angle, optional
+          Tolerance on matching coordinates
+          Only used if chk_sep=True
+        chk_sep : bool, optional
+          Perform coordinate check (expensive)
+        """
+        # Coordinates
+        if chk_sep:
+            testcoord = bool(self.coord.separation(abscomp.coord) < tol)
+        else:
+            testcoord = True
+
+        # Combine
+        test = testcoord
+        # Append?
+        if test:
+            self._components.append(abscomp)
+        else:
+            warnings.warn("Failed add_component test")
+            print('Input absline with wrest={:g} does not match component rules. Not appending'.format(absline.wrest))
+            if not testcoord:
+                print("AbsComp coordinates do not match.  Best to set them")
+
+    def build_table(self):
```

Comment:
I think there should be a generic function in isgm.utils that receives a list of components and creates the table. This method could call that function.